drivers/sony/displays/bravia.cr

```diff
@@ -0,0 +1,269 @@
+require "placeos-driver/interface/powerable"
+require "placeos-driver/interface/muteable"
+require "placeos-driver/interface/switchable"
+
+class Sony::Displays::Bravia < PlaceOS::Driver
+  include Interface::Powerable
+  include Interface::Muteable
+
+  INDICATOR = "\x2A\x53" # *S
+  HASH      = "################"
+  ERROR     = 70
+
+  # Discovery Information
+  tcp_port 20060
+  descriptive_name "Sony Bravia LCD Display"
+  generic_name :Display
+
+  enum Inputs
+    Tv1
+    Tv2
+    Tv3
+    Hdmi1
+    Hdmi2
+    Hdmi3
+    Mirror1
+    Mirror2
+    Mirror3
+    Vga1
+    Vga2
+    Vga3
+
+    def index : Int32
+      to_s.gsub(/[^0-9]/, "").to_i
+    end
+
+    def type_hint : String
+      case self
+      in Tv1, Tv2, Tv3
+        "0000"
+      in Hdmi1, Hdmi2, Hdmi3
+        "1000"
+      in Mirror1, Mirror2, Mirror3
+        "5000"
+      in Vga1, Vga2, Vga3
+        "6000"
+      end
+    end
+
+    def self.from_message?(type_hint, index)
+      type = case type_hint
+             when '0' then "Tv"
+             when '1' then "Hdmi"
+             when '5' then "Mirror"
+             when '6' then "Vga"
+             end
+
+      "#{type}#{index}" if type
+    end
+
+    def to_message : String
+      "#{type_hint}#{(index).to_s.rjust(5, '0')}"
+    end
+  end
+
+  include Interface::InputSelection(Inputs)
+
+  def switch_to(input : Inputs)
+    logger.debug { "switching input to #{input}" }
+    request(Command::Input, input.to_message)
+    self[:input] = input.to_s
+    input?
```

Comment:
A lot of these control methods looks to be running a second command. This should be needed. Responses from control commands will enable you to update state if there was a success response.